src/xcode/ENA/ENA/Source/Services/OTP/OTPService.swift

```diff
@@ -0,0 +1,100 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+protocol OTPServiceProviding {
+	func getValidOTP(ppacToken: PPACToken, completion: @escaping (Result<String, OTPError>) -> Void)
+}
+
+final class OTPService: OTPServiceProviding {
+
+	// MARK: - Init
+
+	init(
+		store: Store,
+		client: Client
+	) {
+		self.store = store
+		self.client = client
+	}
+
+	// MARK: - Overrides
+
+	// MARK: - Protocol OTPServiceProviding
+
+	func getValidOTP(ppacToken: PPACToken, completion: @escaping (Result<String, OTPError>) -> Void) {
+
+		// Check for existing otp. If we have none, create one and proceed.
+		if let token = store.otpToken {
+
+			// We have a token, check now if it is not from the current month or the expirationDate is not reeched
+			let timestamp = token.timestamp
+			if Date() <= token.timestamp {
+				guard Calendar.current.isDate(timestamp, equalTo: Date(), toGranularity: .month) else {
```

Comment:
I'm not sure if a check for month only is enough here. Better also check for .year?

Reply:
good point!